### PR TITLE
#47 Improved publishing and included example of depending on Smart Mastering

### PR DESCRIPTION
**Pull request by rjrudin**
See the minimal-project as an example - and feel free to delete it, I'm not yet proposing that this project have an "examples" directory like ml-gradle does, I just wanted to show an example.

I'm also wondering if we should split out the modules part of SM like for ml-unit-test - i.e. have a smart-mastering-modules subproject. I can see value in an e.g. smart-mastering-java-client subproject as well, given that SM has a number of services associated with it.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -52,57 +54,48 @@ Smart Mastering Core is a community-supported project. Help is available by
...
-maven {
-  url {"https://dl.bintray.com/marklogic-community/Maven/"}
-}
-```
+As this project hasn't been published to the [jcenter](https://bintray.com/bintray/jcenter) repository yet, you'll first
+need to publish a copy of this project to your local Maven repository, which defaults to ~/.m2/repository. 
 
-In the `configurations` section, add a line with `smartMasteringCore` all by
-itself.
+To do so, just run the following command in this directory:
```
Comment (dmcassel) [inline README.md]: Should this say, "To do so, clone this repository and run the following command in the project's directory:"? (just to be clear about required steps)

File: README.md [hunk 2]
```diff
@@ -52,57 +54,48 @@ Smart Mastering Core is a community-supported project. Help is available by
...
-Add this task, which will handle deploying the contents of that zip file:
+    dependencies {
+      mlRestApi "com.marklogic.community:smart-mastering-core:0.1.DEV"
+    }
 
-```
-task deployCore(type: com.marklogic.gradle.task.client.LoadModulesTask) {
-}
-```
+This assumes that the version of the artifacts you published above is 0.1.DEV - if not, just change it to whatever
```
Comment (dmcassel) [inline README.md]: "This assumes that the version of the artifacts you published above is 0.1.DEV. You can find the version number by looking at the `version` property in gradle.properties in your cloned copy of smart-mastering-core."

File: gradle.properties [hunk 3]
```diff
@@ -10,7 +10,8 @@ mlTestRestPort=8042
 # paths to deploy for development
 mlModulePaths=src/main/ml-modules,src/test/ml-modules
 
-pkg_version=0.0.14
+group=com.marklogic.community
+version=0.1.DEV
 
 # the version of ml-gradle that this project uses
 mlGradleVersion=3.6.3
```
Comment (dmcassel) [inline gradle.properties]: does this still need to be here since you added the gradle wrapper?

File: gradle.properties [hunk 4]
```diff
@@ -10,7 +10,8 @@ mlTestRestPort=8042
 # paths to deploy for development
 mlModulePaths=src/main/ml-modules,src/test/ml-modules
 
-pkg_version=0.0.14
+group=com.marklogic.community
+version=0.1.DEV
```
Comment (dmcassel) [inline gradle.properties]: why "0.1.DEV"? while we haven't been real good about semantic versioning yet, that's what we're intending.

File: README.md [hunk 5]
```diff
@@ -127,7 +120,7 @@ Click the `Run Tests` button.
 - add these properties to your `gradle-local.properties`
   - bintray_user
   - bintray_key
-- change the `pkg_version` property in `gradle.properties` to the new version number
+- change the `version` property in `gradle.properties` to the new version number
 - `gradle bintrayUpload`
```
Comment (dmcassel) [inline README.md]: might as well change to `./gradlew bintryUpload` for consistency

**Comment by dmcassel** [comment]
@rjrudin thanks for sending this

File: README.md (hunk 1 above)
Comment (rjrudin) [reply README.md]: Yup, will change

File: gradle.properties (hunk 4 above)
Comment (rjrudin) [reply gradle.properties]: I didn't want to use a real version number here as I wasn't sure what to propose. I'm open for anything, I just didn't want to lay claim to e.g. "0.1.0".

File: README.md (hunk 5 above)
Comment (rjrudin) [reply README.md]: Done

File: gradle.properties (hunk 4 above)
Comment (dmcassel) [reply gradle.properties]: Let's go with 0.0.15 for now. I figure we'll do a release at the end of this sprint (next week) and will probably bump it to 0.1.0 then, but for now, let's keep it in line with the trend.

**Pull request merged**